src/chartsymbols.cpp

```diff
@@ -289,7 +289,7 @@ void ChartSymbols::ProcessLookups( pugi::xml_node &node )
             else if( !strcmp( lookupNode.name(), "attrib-code") ) {
                 int nc = strlen(nodeText);
                 char *attVal = (char *)calloc(nc+2, sizeof(char));
-                strncpy(attVal, nodeText, nc);
+                memcpy(attVal, nodeText, nc);
                 
                 if( attVal[6] == '\0')
                     attVal[6] = ' ';
```

Comment:
A pre-existing issue:
What happens here if e.g. nc=1? Write outside the allocated string - potentially corrupting unrelated memory?

Reply:
I don't folow you here... if nc == 1 the allocated buffer is three (nc + 2)  bytes long and there should not be a problem. Or?

BTW: Nice to have fresh eyes looking at this...

Reply:
Ah... got it. We should ensure that the buffer is at least 7 bytes, right?

Reply:
I'm not sure about the surrounding context and what behavior is wanted by the check, is it to extend strings with 6 chars with a space? Shorted input should probably either be rejected or space padded? 

However at least the write outside the array should be avoided.

Reply:
> I'm not sure about the surrounding context

Neither am I. And what worries me is this obvious bug isn't alone. We need to review this whole file, I guess -- to get the context, and check for more. 

Good catch!